app/uk.gov.hmrc.helptosavestridefrontend/controllers/SessionBehaviour.scala

```diff
@@ -47,18 +49,18 @@ trait SessionBehaviour {
       ).flatMap(identity)
 
   def checkSession(noSessionData:         ⇒ Future[Result],
-                   whenEligible:          (Eligible, Boolean, NSIPayload) ⇒ Future[Result] = (_, _, _) ⇒ SeeOther(routes.StrideController.customerEligible().url),
-                   whenIneligible:        (Ineligible, NSIPayload) ⇒ Future[Result]        = (_, _) ⇒ SeeOther(routes.StrideController.customerNotEligible().url),
-                   whenAlreadyHasAccount: NSIPayload ⇒ Future[Result] = _ ⇒ SeeOther(routes.StrideController.accountAlreadyExists().url)
+                   whenEligible:          (Eligible, Boolean, NSIPayload, Option[AccountReferenceNumber]) ⇒ Future[Result] = (_, _, _, _) ⇒ SeeOther(routes.StrideController.customerEligible().url),
+                   whenIneligible:        (Ineligible, NSIPayload, Option[AccountReferenceNumber]) ⇒ Future[Result]        = (_, _, _) ⇒ SeeOther(routes.StrideController.customerNotEligible().url),
+                   whenAlreadyHasAccount: (NSIPayload, Option[AccountReferenceNumber]) ⇒ Future[Result]                    = (_, _) ⇒ SeeOther(routes.StrideController.accountAlreadyExists().url)
   )(implicit request: Request[_]): Future[Result] =
     checkSessionInternal(
       noSessionData,
 
       htsSession ⇒
         htsSession.userInfo match {
-          case e: Eligible       ⇒ whenEligible(e, htsSession.detailsConfirmed, htsSession.nSIUserInfo)
-          case i: Ineligible     ⇒ whenIneligible(i, htsSession.nSIUserInfo)
-          case AlreadyHasAccount ⇒ whenAlreadyHasAccount(htsSession.nSIUserInfo)
+          case e: Eligible       ⇒ whenEligible(e, htsSession.detailsConfirmed, htsSession.nSIUserInfo, htsSession.accountNumber)
+          case i: Ineligible     ⇒ whenIneligible(i, htsSession.nSIUserInfo, htsSession.accountNumber)
```

Comment:
does it make sense to have the account number here within the Ineligible case?

Reply:
for manual account creation we could have it

Reply:
oh yeah, ok I'm happy with that